src/Core/OrganizationFeatures/AuthorizationHandlers/CollectionUserAuthorizationHandler.cs

```diff
@@ -0,0 +1,207 @@
+
+        // List of collections the user is performing the operation on
+        var targetCollections = await _collectionRepository.GetManyByManyIdsAsync(distinctTargetCollectionIds);
+
+        // A target collection does not exist, fail the requirement
+        if (targetCollections.Count != distinctTargetCollectionIds.Count)
+        {
+            context.Fail();
+            return;
+        }
+
+        // TODO: Add check for future organization Collection Management setting here and skip the next check if it's enabled
+
+        // List of collections the user is allowed to manage
+        var manageableCollections = (await _collectionRepository.GetManyByUserIdAsync(_currentContext.UserId.Value)).Where(c => c.Manage).ToList();
+
+        // Any target collection is not in the list of manageable collections, fail the requirement
+        if (targetCollections.Any(tc => !manageableCollections.Exists(c => c.Id == tc.Id)))
+        {
+            context.Fail();
+            return;
+        }
+
+        var distinctTargetUserIds = resource.Select(c => c.OrganizationUserId).Distinct().ToList();
+
+        // List of users the user is performing the operation on
+        var targetUsers = await _organizationUserRepository.GetManyAsync(distinctTargetUserIds);
+
+        // A target user does not exist, fail the requirement
+        if (targetUsers.Count != distinctTargetUserIds.Count)
+        {
+            context.Fail();
+            return;
+        }
+
+        foreach (var targetUser in targetUsers)
+        {
+            var targetCollectionsForUser = from c in targetCollections
+                                           join cu in resource on c.Id equals cu.CollectionId
+                                           where cu.OrganizationUserId == targetUser.Id
+                                           select c;
+
+            // Target user is not in the same org as a collection they're being assigned to, fail the requirement
+            if (targetCollectionsForUser.Any(tc => tc.OrganizationId != targetUser.OrganizationId))
+            {
+                context.Fail();
+                return;
+            }
+        }
+
+        context.Succeed(requirement);
+    }
+
+    /// <summary>
+    /// Ensure the acting user can delete the requested <see cref="CollectionUser"/> resource(s).
+    /// </summary>
+    /// <remarks>
+    /// Checks that the following are all true:
+    /// - The target collection(s) exists 
+    /// - The acting user is an owner/admin AND the collection management setting is enabled
+    ///   OR
+    ///   The target collection(s) is manageable by the acting user. 
+    /// </remarks>
+    private async Task CanDeleteAsync(AuthorizationHandlerContext context,
+        CollectionUserOperationRequirement requirement, ICollection<CollectionUser> resource)
+    {
+        if (!_currentContext.UserId.HasValue)
+        {
+            context.Fail();
+            return;
+        }
```

Comment:
Will this be the case for all AuthorizationHandlers? What does it mean if the currentContext doesn't have a userId set?

In any case, this can go in the `HandleRequirementAsync` method to avoid repeating it in each private method.

Reply:
I believe it will be the most common case for AuthorizationHandlers, at least initially. All expected use cases for the handlers in this ticket will have a userId as they will be triggered by controller endpoints used by users via the Web client.

I would expect us to adapt the handler to account for any use cases where a `userId` is not expected (e.g. another type of authenticated subject is making a request that doesn't have a userId). But, for now, I think we're safe to assume a `userId` must be present to authorize successfully. 

And yep, we can move the check to the `HandleRequirementAsync` method to reduce repetition (and redundant tests).